project-base/storefront/components/Layout/Header/AutocompleteSearch/AutocompleteSearch.tsx

```diff
@@ -1,143 +1,96 @@
 import { Close } from 'components/Basic/Icon/IconsSvg';
 import { SearchInput } from 'components/Forms/TextInput/SearchInput';
-import { desktopFirstSizes } from 'components/Theme/mediaQueries';
 import { useAutocompleteSearchQueryApi } from 'graphql/generated';
 import { getInternationalizedStaticUrls } from 'helpers/getInternationalizedStaticUrls';
 import { useGtmAutocompleteResultsViewEvent } from 'gtm/hooks/useGtmAutocompleteResultsViewEvent';
 import { useDebounce } from 'hooks/helpers/useDebounce';
 import useTranslation from 'next-translate/useTranslation';
-import { useGetWindowSize } from 'hooks/ui/useGetWindowSize';
-import { useResizeWidthEffect } from 'hooks/ui/useResizeWidthEffect';
 import { useDomainConfig } from 'hooks/useDomainConfig';
 import dynamic from 'next/dynamic';
 import { useRouter } from 'next/router';
-import { ChangeEventHandler, useCallback, useEffect, useRef, useState } from 'react';
+import { useState } from 'react';
 import { twJoin } from 'tailwind-merge';
+import { Overlay } from 'components/Basic/Overlay/Overlay';
 
-const Autocomplete = dynamic(() => import('./Autocomplete').then((component) => component.Autocomplete));
+const AutocompleteSearchPopup = dynamic(() =>
+    import('./AutocompleteSearchPopup').then((component) => component.AutocompleteSearchPopup),
+);
 
 export const MINIMAL_SEARCH_QUERY_LENGTH = 3 as const;
-const TEST_IDENTIFIER = 'layout-header-search-autocomplete-input';
 
 export const AutocompleteSearch: FC = () => {
     const router = useRouter();
     const [autocompleteSearchQueryValue, setAutocompleteSearchQueryValue] = useState('');
     const debouncedAutocompleteSearchQuery = useDebounce(autocompleteSearchQueryValue, 200);
-    const [hasAutocompleteSearchFocus, setAutocompleteSearchFocus] = useState(false);
+    const { url } = useDomainConfig();
+    const [searchUrl] = getInternationalizedStaticUrls(['/search'], url);
+    const { t } = useTranslation();
+
+    const isWithValidSearchQuery = autocompleteSearchQueryValue.length >= MINIMAL_SEARCH_QUERY_LENGTH;
+
     const [{ data: autocompleteSearchData, fetching: areSearchResultsLoading }] = useAutocompleteSearchQueryApi({
         variables: {
             search: debouncedAutocompleteSearchQuery,
             maxCategoryCount: AUTOCOMPLETE_CATEGORY_LIMIT,
             maxProductCount: AUTOCOMPLETE_PRODUCT_LIMIT,
         },
-        pause: debouncedAutocompleteSearchQuery.length < MINIMAL_SEARCH_QUERY_LENGTH,
+        pause: !isWithValidSearchQuery,
         requestPolicy: 'network-only',
     });
 
-    const autocompleteSearchInRef = useRef<HTMLDivElement>(null);
-    const { url } = useDomainConfig();
-    const [searchUrl] = getInternationalizedStaticUrls(['/search'], url);
-    const { t } = useTranslation();
-    const [isDesktop, setIsDesktop] = useState(false);
-    const { width } = useGetWindowSize();
+    const isSearchResultsPopupVisible = isWithValidSearchQuery && !!autocompleteSearchData;
```

Comment:
This condition makes the popup disappear and appear after every keystroke

Reply:
This is still happening (though its not right after the keystroke but when data is being loaded). According to Peta this is a bug. Please, take a look at it.

Reply:
this one is solved, nice :)